review/fileToReview.js

```diff
@@ -1,73 +1,88 @@
 
-var domain = "bank.local.fr"
+const domain = "bank.local.fr"
 
 /**
  * @description Fetch transactions recursively
- * @param {string} fromDate The maximum date of transactions to return
- * @param {string} authorization Authorization header to authent the user
- * @param {jws} jws Jws token, mandatory if we get one from login request
- * @param {Number} id Account id
+ * @param {String} fromDate The maximum date of transactions to return
+ * @param {String} authorization Authorization header to authent the user
+ * @param {String} jwt jwt token, mandatory if we get one from login request
+ * @param {Number} account_id Account id
  * @param {Number} page Number of the page
- * @param {Object} previousTransactions Previous page of transactions (To ckeck for dupes)
- * @return {Object} All transactions available on the page
+ * @param {Object} allTransactions Previous page of transactions (To ckeck for dupes)
+ * @return {Promise<Array>} All transactions available on the page
  */
-async function fetchTransactions(fromDate, authorization, jws = null, id, page, previousTransactions) {
-	console.log(`--- Fetch Trasactions page n°${page} ---`);
-	try {
-    var headers = {"Authorisation":  authorization }
-
-    if (jws) {
-      headers = {
-        "Authorisation": authorization,
-        "jws": jws,
-        "Content-type": "application/json",
-        "Accept": "application/json"
-      }
-    } else {
-      headers = {
-        "Authorisation": authorization,
-        "Content-type": "application/json",
-        "Accept": "application/json",
-      }
+async function fetchTransactions(fromDate, authorization, jwt = null, account_id, page = 0, allTransactions = []) {
```

Comment:
if im not wrong , the goal is to return all transactions in all pages ?
So the edits here is we have a shared array and each page we push new transactions to it 
and at end we return the array